File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/usecases/photos/PhotoFilterUseCaseImpl.java
package nl.acidcats.tumblrlikes.core.usecases.photos;

import javax.inject.Inject;

import nl.acidcats.tumblrlikes.core.constants.FilterType;
import nl.acidcats.tumblrlikes.core.repositories.PhotoDataRepository;
import rx.Observable;

/**
 * Created on 19/10/2018.
 */
public class PhotoFilterUseCaseImpl implements PhotoFilterUseCase {

    private PhotoDataRepository _photoDataRepository;

    @Inject
    public PhotoFilterUseCaseImpl(PhotoDataRepository photoDataRepository) {
        _photoDataRepository = photoDataRepository;
    }

    @Override
    public Observable<Boolean> storeFilterSelection(FilterType filterType) {
        _photoDataRepository.setFilterType(filterType);

        return Observable.just(true);
    }

    @Override
    public Observable<FilterType> getSelectedFilterType() {
        return Observable.just(_photoDataRepository.getFilterType());
    }
}

File: datalib/src/main/kotlin/nl/acidcats/tumblrlikes/core/usecases/photos/PhotoFilterUseCaseImpl.kt
package nl.acidcats.tumblrlikes.core.usecases.photos

import nl.acidcats.tumblrlikes.core.constants.FilterType
import nl.acidcats.tumblrlikes.core.repositories.PhotoDataRepository
import rx.Observable
import javax.inject.Inject

/**
 * Created on 24/10/2018.
 */
class PhotoFilterUseCaseImpl @Inject constructor(private val photoDataRepository: PhotoDataRepository) : PhotoFilterUseCase {

    override fun getSelectedFilterType(): Observable<FilterType> {
        return Observable.just(photoDataRepository.getFilterType())
    }

    override fun storeFilterSelection(filterType: FilterType): Observable<FilterType> {
        photoDataRepository.setFilterType(filterType)

        return Observable.just(filterType)
    }
}